lib/executor/constant_arrival_rate.go

```diff
@@ -196,9 +199,24 @@ func (car *ConstantArrivalRate) Init(ctx context.Context) error {
 	// with no work, as determined by their config's HasWork() method.
 	et, err := car.BaseExecutor.executionState.ExecutionTuple.GetNewExecutionTupleFromValue(car.config.MaxVUs.Int64)
 	car.et = et
+	start, offsets, lcd := et.GetStripedOffsets()
+	car.segIdx = lib.NewSegmentedIndex(start, lcd, offsets)
+
 	return err
 }
 
+// incrGlobalIter increments the global iteration count for this executor,
+// taking into account the configured execution segment.
+func (car *ConstantArrivalRate) incrGlobalIter() {
+	car.segIdx.Next()
+	atomic.StoreUint64(car.globalIter, uint64(car.segIdx.GetUnscaled()))
```

Comment:
This is logically racy as there is nothing preventing 2 goroutines to call `car.segIdx.Next()` and then `atomic.StoreUint64` and the one who got the "first" values from `segIdx.Next` to store it second, effectively having the previous value in `car.globalIter`.

Same for the `shared-iterations`

Reply:
Let me know if it's as simple as 8973a841.

I wanted to remove the use of `atomic`, but we still need it because `globalIter` needs to remain a pointer since `incrGlobalIter` is passed to the VU `State` which is where it's being called (this is part of the refactor to fix how `globalIter` works).